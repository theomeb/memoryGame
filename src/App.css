.memory {
  display: flex;
  flex-wrap: wrap;
  width: 300px;
  margin: auto;
  user-select: none;
  border-style: solid;
  border-color: rgba(61, 193, 207, 0.651)
}