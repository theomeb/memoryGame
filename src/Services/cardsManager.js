import shuffle from 'lodash.shuffle'

const SIDE = 6
export const SYMBOLS = '😀🎉💖🎩🐶🐱🦄🐬🌍🌛🌞💫🍎🍌🍓🍐🍟🍿'

export const  generateCards = function () {
    const result = []
    const size = SIDE * SIDE
    const candidates = shuffle(SYMBOLS)
    while (result.length < size) {
      const card = candidates.pop()
      result.push(card, card)
  	}
   	return shuffle(result)
  };



