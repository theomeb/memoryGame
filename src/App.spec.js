import { expect } from 'chai'
import React from 'react'
import { shallow } from 'enzyme'

import App from './App'
import { SYMBOLS } from './Services/cardsManager'
import GuessCount from './GuessCount'
import Card from './Card'
import sinon from 'sinon'



describe('<App />', () => {
    it('renders without crashing', () => {
      const wrapper = shallow(<App />)
    })

    it('contains a zero-guess counter', () => {
      const wrapper = shallow(<App />)
      expect(wrapper).to.contain(<GuessCount guesses={0} />)
    })

    it('has 36 cards', () => {
      const wrapper = shallow(<App />)
      expect(wrapper.find('Card')).to.have.length(36)
    })

    //// Renvoie tjrs un snapshot différent...
    // it('should match its reference snapshot', () => {
    //     const mockSymbols = [...SYMBOLS.repeat(2)];
    //     jest.mock('./Services/cardsManager', () => ({
    //       generateCards: jest.fn(() => [...mockSymbols]),
    //     }))

    //     try {
    //       const wrapper = shallow(<App />)
    //       expect(wrapper).to.matchSnapshot()
    //     } finally {
          
    //     }
    // })

  describe('<Card/>', () => {
        it('should trigger its `onClick` prop when clicked', () => {
            const onClick = sinon.spy()
            const wrapper = shallow(
              <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
            )
            wrapper.simulate('click')
            expect(onClick).to.have.been.calledWith(0)
        })

        it('should match its reference snapshot', () => {
            const onClick = sinon.spy()
            const wrapper = shallow(
            <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
            )
            expect(wrapper).to.matchSnapshot()
        })
    })

})

