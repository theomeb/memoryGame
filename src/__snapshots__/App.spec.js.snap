// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> <Card/> should match its reference snapshot 1`] = `
<div
  className="card hidden"
  onClick={[Function]}
>
  <span
    className="symbol"
  >
    ‚ùì
  </span>
</div>
`;
